woocommerce/woocommerce-ios
# [JITM] Support in other screens infrastructure

### Description

We want to refactor here the JITM implementation to add new placements in screens other than the dashboard. For that, we move the JITM retrieval logic out of `DashboardViewModel` into its own class, so it can be reused for new placements. Furthermore, we also add an `enum` to gather the placement screens, hiding the map between them and the message path string inside the new `JustInTimeMessagesProvider`. 

Lastly, we convert the logic to `async/await`, making it easier to load the new messages and synchronize the products onboarding priority logic. In order to ensure that UI code is running on the main thread, we use  `Task` with @ `MainActor`.

### Testing instructions

Ensure that the JITM in the Dashboard is still working as expected. Be sure to cover all cases: showing the JITM, dismissal, refresh, and orientation changes. Furthermore, unit tests should pass.

### Screenshots



---
